app/com/arpnetworking/metrics/portal/alerts/impl/DailyPartitionCreator.java

```diff
@@ -0,0 +1,283 @@
+                    scheduleOffset,
+                    lookahead
+                )
+        );
+    }
+
+    /**
+     * Ask the actor referenced by {@code ref} to start execution.
+     * <p>
+     * This will execute partition creation exactly once and then periodically thereafter using the actor's props.
+     *
+     * @param ref an {@code DailyPartitionCreator}.
+     * @param timeout timeout for the operation
+     * @throws ExecutionException if an exception was thrown during execution.
+     * @throws InterruptedException if the actor does not reply within the allotted timeout, or if the actor thread was
+     * interrupted for other reasons.
+     */
+    public static void start(
+            final ActorRef ref,
+            final Duration timeout
+    ) throws ExecutionException, InterruptedException {
+        Patterns.ask(
+                ref,
+                START_TICKING,
+                timeout
+        )
+                .thenCompose(reply -> {
+                    @SuppressWarnings("unchecked") final Optional<Throwable> o = (Optional<Throwable>) reply;
+                    final CompletableFuture<Void> future = new CompletableFuture<>();
+                    o.ifPresent(future::completeExceptionally);
+                    future.complete(null);
+                    return future;
+                })
+                .toCompletableFuture()
+                .get();
+    }
+
+    /**
+     * Ask the actor referenced by {@code ref} to stop execution.
+     *
+     * @param ref an {@code DailyPartitionCreator}.
+     * @param timeout timeout for the operation
+     * @throws ExecutionException if an exception was thrown during execution.
+     * @throws InterruptedException if the actor does not stop within the allotted timeout, or if the actor thread was
+     * interrupted for other reasons.
+     */
+    public static void stop(final ActorRef ref, final Duration timeout) throws ExecutionException, InterruptedException {
+        Patterns.gracefulStop(ref, timeout).toCompletableFuture().get();
+    }
+
+    @Override
+    public void postStop() throws Exception {
+        super.postStop();
+        LOGGER.info().setMessage("Actor was stopped")
+                .addData("schema", _schema)
+                .addData("table", _table)
+                .addData("lookahead", _lookahead)
+                .log();
+    }
+
+    @Override
+    public Receive createReceive() {
+        return new ReceiveBuilder()
+                .matchEquals(START_TICKING, msg -> startTicking())
+                .matchEquals(TICK, msg -> tick())
+                .matchEquals(EXECUTE, msg -> execute())
+                .build();
+    }
+
+    private void recordCounter(final String metricName, final long value) {
+        final String fullMetric = String.format("partition_creator/%s/%s", _table, metricName);
+        _periodicMetrics.recordCounter(fullMetric, value);
+    }
+
+    // Message handlers
+
+    private void startTicking() {
```

Comment:
Moving this to onStart would remove the statefulness of the actor entirely; e.g. if it exists then it's running

Reply:
(marking as resolved since this outcome is tied to the other comment, but feedback makes sense)